docs/src/main/tut/GettingStarted.md

```diff
@@ -0,0 +1,81 @@
+# Getting started
+We are going to manipulate some `TypedDataset`s through a small toy example.
+Before starting the tutorial, let's import/define a couple of things.
+```tut:book
+import org.apache.spark.{SparkConf, SparkContext}
+import org.apache.spark.sql.SparkSession
+import frameless.functions.aggregate._
+import frameless.TypedDataset
+
+val conf = new SparkConf().setMaster("local[*]").setAppName("frameless repl").set("spark.ui.enabled", "false")
+val spark = SparkSession.builder().config(conf).appName("REPL").getOrCreate()
+implicit val sqlContext = spark.sqlContext
+spark.sparkContext.setLogLevel("WARN")
+
+import spark.implicits._
+``` 
+## Creating TypedDataset instances
+We start by defining a simple case class that will be the basis of our examples.
+```tut:book
+case class Appartment(city: String, surface: Int, price: Double)
+``` 
+And let's define a few `Appartment` instances:
+```tut:book
+val appartments = Seq(
+    Appartment("Paris", 50, 300000.0),
+    Appartment("Paris", 100, 450000.0),
+    Appartment("Paris", 25, 250000.0),
+    Appartment("Lyon", 83, 200000.0),
+    Appartment("Lyon", 45, 133000.0),
+    Appartment("Nice", 74, 325000.0)
+    )
+```
+We can create our `TypedDataset` directly :
+```tut:book
+val appartmentsTypedDS = TypedDataset.create(appartments)
+```
+We can also create it from an existing `Dataset` :
+```tut:book
+val appartmentsDS = spark.createDataset(appartments)
+val appartmentsTypedDS = TypedDataset.create(appartmentsDS)
+```
+## Typesafe column referencing
+Selecting a column from our `TypedDataset` is done in this manner : 
+```tut:book
+val cities: TypedDataset[String] = appartmentsTypedDS.select(appartmentsTypedDS('city))
+```
+This is completely safe, for instance suppose we made a typo :
+```tut:book:fail
+appartmentsTypedDS.select(appartmentsTypedDS('citi))                                          ^
+```
+This gets caught at compile time.
+With `Dataset` the error would have been caught at runtime.
+```tut:book:fail
+appartmentsDS.select('citi)
+``` 
+Let us now try to get the price by surface unit : 
+```tut:book:fail
+val priceBySurfaceUnit = appartmentsTypedDS.select(appartmentsTypedDS('price)/appartmentsTypedDS('surface))                                                                          ^
+```
+Argh! Looks like we can't divide a `TypedColumn` of `Double` by `Int`. 
```

Comment:
We actually have a way to do this. If you want all the automated widenings supported by Spark you just have to include.
```scala
import frameless.implicits.widen._
```